src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -438,6 +438,37 @@ JNI_METHOD(void, setDeviceAttestationDelegate)
     }
 }
 
+JNI_METHOD(void, setAttestationTrustStoreDelegate)
+(JNIEnv * env, jobject self, jlong handle, jobject attestationTrustStoreDelegate)
+{
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    ChipLogProgress(Controller, "setAttestationTrustStoreDelegate() called");
+
+    if (attestationTrustStoreDelegate != nullptr)
+    {
+        wrapper->ClearAttestationTrustStoreBridge();
+        jobject attestationTrustStoreDelegateRef = env->NewGlobalRef(attestationTrustStoreDelegate);
+        wrapper->SetAttestationTrustStoreBridge(new AttestationTrustStoreBridge(attestationTrustStoreDelegateRef));
+        VerifyOrExit(wrapper->GetAttestationTrustStoreBridge() != nullptr, err = CHIP_ERROR_NO_MEMORY);
```

Comment:
Please allocate separately so we never get `wrapper->SetAttestationTrustStoreBridge(nullptr)` due to out of memory, since this would replace the previous one that was properly allocated if there was one,